packages/tradershub/src/components/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { ReactElement } from 'react';
+import { useEventListener } from 'usehooks-ts';
+import { qtMerge } from '@deriv/quill-design';
+import { useModal } from '../ModalProvider';
+import DialogAction from './DialogAction';
+import DialogContent from './DialogContent';
+import DialogHeader from './DialogHeader';
+
+/**
+ * Type for the Dialog children
+ * @typedef TDialogChildren
+ */
+type TDialogChildren =
+    | ReactElement<typeof DialogAction>
+    | ReactElement<typeof DialogContent>
+    | ReactElement<typeof DialogHeader>;
+
+/**
+ * Type for the Dialog component props
+ * @typedef TDialog
+ * @property {TDialogChildren | TDialogChildren[]} children - Children nodes
+ * @property {string} [className] - Optional CSS class name
+ * @property {boolean} [shouldPreventCloseOnEscape] - Optional flag to prevent closing the dialog on Escape key press
+ */
+type TDialog = {
+    children: TDialogChildren | TDialogChildren[];
+    className?: string;
+    shouldPreventCloseOnEscape?: boolean;
+};
+
+/**
+ * Dialog component
+ * @param {TDialog} props - The properties that define the Dialog component.
+ * @returns {JSX.Element} The Dialog component.
+ *
+ * @example
+ * ```tsx
+ * <Dialog>
+ *   <Dialog.Header title='Dialog title'/>
+ *   <Dialog.Content>Dialog content goes here</Dialog.Content>
+ *   <Dialog.Action align='center'>
+ *      <Button>Primary action</Button>
+ *   </Dialog.Action>
+ * </Dialog>
+ * ```
+ */
+const Dialog = ({ children, className, shouldPreventCloseOnEscape = false }: TDialog) => {
```

Comment:
We will use this for multi step modals later on, so isn't it better to keep the Modal instead of Dialog?